mesonbuild/interpreter.py

```diff
@@ -2996,31 +3017,46 @@ def _find_cached_dep(self, name, kwargs):
         # Check if we want this as a build-time / build machine or runt-time /
         # host machine dep.
         for_machine = self.machine_from_native_kwarg(kwargs)
-
         identifier = dependencies.get_dep_identifier(name, kwargs)
-        cached_dep = self.coredata.deps[for_machine].get(identifier)
+        wanted_vers = mesonlib.stringlistify(kwargs.get('version', []))
+
+        cached_dep = self.build.dependency_overrides[for_machine].get(identifier)
         if cached_dep:
+            # We don't implicitly override not-found dependencies, but user could
+            # have explicitly called meson.override_dependency() with a not-found
+            # dep.
             if not cached_dep.found():
                 mlog.log('Dependency', mlog.bold(name),
                          'found:', mlog.red('NO'), mlog.blue('(cached)'))
```

Comment:
Is this really "cached"? I think this is only reachable for explicitly overridden dependencies. So we should show a sensible error message.

Reply:
Done.